### PR TITLE
ROW: clean up in preparation to hide CharRow & AttrRow

### PR DESCRIPTION
**Pull request by DHowett**
Moving things out of CharRow into ROW helps us hide it as an implementation detail.
This is part one of many.

### Remove RowCellIterator (dead code)

RCI was unused; it was replaced by TextBufferCellIterator shortly after its creation

### Move AttrRowTests to ut_textbuffer from ut_host

It had no reliance on the host.

### CODE REVIEW

**Comment by zadjii-msft** [comment]
@msftbot make sure @miniksa signs off on this

File: src/buffer/out/Row.hpp [hunk 1]
```diff
@@ -36,6 +36,12 @@ class ROW final
 
     size_t size() const noexcept;
 
+    void SetWrapForced(const bool wrap) noexcept;
```
Comment (miniksa) [inline src/buffer/out/Row.hpp:39]: Could just move the getters/setters up here per the other PR and chat recently about inline potential.

File: src/host/ut_host/CharRowBaseTests.cpp [hunk 2]
```diff
@@ -1,517 +0,0 @@
...
-
-#include "../Ucs2CharRow.hpp"
-#include "../Utf8CharRow.hpp"
-
-using namespace WEX::Logging;
-using namespace WEX::TestExecution;
-
-class CharRowBaseTests
-{
-    TEST_CLASS(CharRowBaseTests);
```
Comment (miniksa) [inline src/host/ut_host/CharRowBaseTests.cpp:17]: This was unused/unexecuted?

Comment (DHowett) [reply src/host/ut_host/CharRowBaseTests.cpp:17]: completely

File: src/buffer/out/Row.hpp (hunk 1 above)
Comment (DHowett) [reply src/buffer/out/Row.hpp:39]: no reason not to

File: src/host/ut_host/CharRowBaseTests.cpp (hunk 2 above)
Comment (miniksa) [reply src/host/ut_host/CharRowBaseTests.cpp:17]: wow.

File: src/buffer/out/ut_textbuffer/TextBuffer.Unit.Tests.vcxproj [hunk 3]
```diff
@@ -18,6 +19,9 @@
     </ClCompile>
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\..\..\types\lib\types.vcxproj">
```
Comment (zadjii-msft) [inline src/buffer/out/ut_textbuffer/TextBuffer.Unit.Tests.vcxproj:22]: Do we need to add `types` to the `TARGETLIBS` for this test too?

**Review by zadjii-msft — APPROVED** [review]
Seems straightforward

File: src/buffer/out/ut_textbuffer/TextBuffer.Unit.Tests.vcxproj (hunk 3 above)
Comment (DHowett) [reply src/buffer/out/ut_textbuffer/TextBuffer.Unit.Tests.vcxproj:22]: GOOD CATCH.

Comment (DHowett) [reply src/buffer/out/ut_textbuffer/TextBuffer.Unit.Tests.vcxproj:23]: fixed!

**Review by miniksa — APPROVED** [review]
Lookin' good. Can't wait until the next step!

**Pull request merged**